lib/galaxy/schema/invocation.py

```diff
@@ -214,3 +231,114 @@ class InvocationMessageResponseModel(BaseModel):
+        title="Dataset ID",
+        description="Dataset ID of the workflow step output.",
+    )
+    uuid: Optional[schema.UUID4] = Field(
+        None,
+        title="UUID",
+        description="Universal unique identifier of the workflow step output dataset.",
+    )
+
+
+class InvocationStepCollectionOutput(Model):
+    src: str = Field(
+        literal_to_value(INVOCATION_STEP_COLLECTION_OUTPUT_SRC),
+        title="src",
+        description="The source model of the output.",
+        const=True,
+        mark_required_in_schema=True,
+    )
+    id: DecodedDatabaseIdField = Field(
+        ...,
+        title="Dataset Collection ID",
+        description="Dataset Collection ID of the workflow step output.",
+    )
+
+
+class InvocationStep(Model):
+    """Information about Workflow Invocation Step"""
+
+    model_class: schema.INVOCATION_STEP_MODEL_CLASS = schema.ModelClassField(schema.INVOCATION_STEP_MODEL_CLASS)
+    id: DecodedDatabaseIdField = schema.EntityIdField
+    update_time: Optional[datetime] = schema.UpdateTimeField
+    job_id: Optional[DecodedDatabaseIdField] = Field(
+        title="Job ID",
+        description="The encoded ID of the job associated with this workflow invocation step.",
+    )
+    workflow_step_id: DecodedDatabaseIdField = Field(
+        ...,
+        title="Workflow step ID",
+        description="The encoded ID of the workflow step associated with this workflow invocation step.",
+    )
+    subworkflow_invocation_id: DecodedDatabaseIdField = Field(
+        title="Subworkflow invocation ID",
+        description="The encoded ID of the subworkflow invocation.",
+    )
+    state: InvocationStepState = Field(
+        ...,
+        title="State of the invocation step",
+        description="Describes where in the scheduling process the workflow invocation step is.",
+    )
+    action: bool = InvocationStepAction
+    order_index: int = Field(
+        ...,
+        title="Order index",
+        description="The index of the workflow step in the workflow.",
+    )
+    workflow_step_label: str = Field(
+        title="Step label",
+        description="The label of the workflow step",
+    )
+    workflow_step_uuid: Optional[schema.UUID4] = Field(
+        None,
+        title="UUID",
+        description="Universal unique identifier of the workflow step.",
+    )
+    outputs: Dict[str, InvocationStepOutput] = Field(
+        {},
+        title="Outputs",
+        description="The outputs of the workflow invocation step.",
+    )
+    output_collections: Dict[str, InvocationStepCollectionOutput] = Field(
+        {},
+        title="Output collections",
+        description="The dataset collection outputs of the workflow invocation step.",
+    )
+    jobs: List[schema.JobBaseModel] = Field(
+        [],
+        title="Jobs",
+        description="Jobs associated with the workflow invocation step.",
+    )
+
+
+class InvocationUpdatePayload(Model):
+    action: bool = InvocationStepAction
```

Comment:
I would rename `InvocationStepAction` to `InvocationStepActionField` (or something similar) to make clear that is a Field definition otherwise it can be confused with some real "action". I know... naming is the hardest part :laughing: